rules/private/phases/phase_semanticdb.bzl

```diff
@@ -33,7 +31,13 @@ def phase_semanticdb(ctx, g):
     outputs = []
 
     for source in ctx.files.srcs:
-        if source.extension == "scala":
+        # Generated or external files will have the output directory (beginning with `bazel-out`) in
+        # their paths, which we don't want because it isn't guaranteed to be consistent
+        if (
+            source.extension == "scala" and \
+            source.is_source and \
+            source.owner.workspace_name == ctx.label.workspace_name
```

Comment:
since I'm seeing the word workspace here, is this bazel 8 compatible?

Reply:
Actually, it's not. Good catch. It seems it was deprecated in favor of the more accurately named `repo_name`:
https://bazel.build/rules/lib/builtins/Label#workspace_name

I'll correct this.